osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/infra/messaging/BaseMessageProcessor.java

```diff
@@ -0,0 +1,205 @@
+     * The MessageProcessor Singleton is added to the Map of MessageProcessors.
+     */
+    @PostConstruct
+    public void init() {
+        this.iec60870RequestMessageProcessorMap.addMessageProcessor(this.messageType, this);
+    }
+
+    @Override
+    public void processMessage(final ObjectMessage message) throws JMSException {
+        LOGGER.info("Processing get health status request message in new code...");
+
+        MessageMetadata messageMetadata = null;
+        try {
+            messageMetadata = MessageMetadata.fromMessage(message);
+
+            final BaseResponseEventListener eventListener = this.createEventListener(messageMetadata);
+            final DeviceConnection deviceConnection = this.iec60870DeviceService.connectToDevice(
+                    messageMetadata.getDeviceIdentification(), messageMetadata.getIpAddress(), eventListener);
+
+            this.process(messageMetadata, deviceConnection);
+        } catch (final ProtocolAdapterException e) {
+            this.handleError(messageMetadata, e);
+        } catch (final Exception e) {
+            LOGGER.error("UNRECOVERABLE ERROR, unable to read ObjectMessage instance, giving up.", e);
+            return;
+        }
+    }
+
+    /**
+     * Create a new instance of the event listener for this message processor.
+     *
+     * @param messageMetadata
+     *            MessageMetaData to pass to the new event listener.
+     * @return The created event listener.
+     */
+    private BaseResponseEventListener createEventListener(final MessageMetadata messageMetadata) {
+
+        Constructor<? extends BaseResponseEventListener> constructor;
+        try {
+            constructor = this.responseEventListener.getConstructor(MessageMetadata.class, ResponseMessageSender.class,
+                    LogItemRequestMessageSender.class);
+
+            return constructor.newInstance(messageMetadata, this.responseMessageSender,
+                    this.iec60870LogItemRequestMessageSender);
+
+        } catch (NoSuchMethodException | SecurityException | InstantiationException | IllegalAccessException
+                | IllegalArgumentException | InvocationTargetException e) {
+
+            throw new IllegalStateException("Failed to create an instance for " + this.responseEventListener.getName(),
+                    e);
+        }
+
+    }
+
+    public abstract void process(final MessageMetadata messageMetadata, final DeviceConnection deviceConnection)
+            throws ProtocolAdapterException;
+
+    protected void printDomainInfo(final RequestMessageData requestMessageData) {
+        LOGGER.info("Calling DeviceService function: {} for domain: {} {}", requestMessageData.getMessageType(),
+                requestMessageData.getDomain(), requestMessageData.getDomainVersion());
+    }
+
+    protected void handleError(final MessageMetadata messageMetadata, final ProtocolAdapterException e)
+            throws JMSException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'javax.jms.JMSException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

Reply:
The JMSException can be thrown by a child of the BaseMessageProcessor. Because it's a checked exception, it has to be thrown by the method in the base class as well.